File: app/src/main/java/com/sixbeeps/uniquity/data/UnicodeGroupWithCharacters.java
package com.sixbeeps.uniquity.data;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class UnicodeGroupWithCharacters {
    @Embedded public UnicodeGroup group;
    @Relation(
            parentColumn = "name",
            entityColumn = "codepoint"
    )
    public List<UnicodeCharacter> characters;
}

File: app/src/main/java/com/sixbeeps/uniquity/data/UnicodeGroupWithCharacters.kt
package com.sixbeeps.uniquity.data

import androidx.room.Embedded
import androidx.room.Relation

class UnicodeGroupWithCharacters {
    @Embedded
    var group: UnicodeGroup? = null

    @Relation(parentColumn = "name", entityColumn = "codepoint")
    var characters: MutableList<UnicodeCharacter?>? = null
}
